drivers/triton/triton.go

```diff
@@ -0,0 +1,573 @@
+		},
+	}
+	u := d.GetHttpsURL()
+
+	resp, err := client.Get(fmt.Sprintf("%s/_ping", u))
+	if err != nil {
+		return state.Error, nil
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if string(body[:]) == "OK" {
+		return state.Running, nil
+	}
+	return state.None, nil
+}
+
+// Kill stops a host forcefully
+func (d *Driver) Kill() error {
+	return fmt.Errorf("Kill is not available for the triton driver")
+}
+
+// PreCreateCheck allows for pre-create operations to make sure a driver is ready for creation
+func (d *Driver) PreCreateCheck() error {
+	return nil
+}
+
+// Remove a host
+func (d *Driver) Remove() error {
+	return nil
+}
+
+// Restart a host. This may just call Stop(); Start() if the provider does not
+// have any special restart behaviour.
+func (d *Driver) Restart() error {
+	return fmt.Errorf("Restart is not available for the triton driver")
+}
+
+// SetConfigFromFlags configures the driver with the object that was returned
+// by RegisterCreateFlags
+func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	d.CloudApiURL = flags.String("triton-url")
+	d.Account = flags.String("triton-account")
+	d.PrivateKey = flags.String("triton-key")
+
+	if d.CloudApiURL == "" {
+		return fmt.Errorf("You must specify the cloudapi url using --triton-url")
+	}
+	if d.Account == "" {
+		return fmt.Errorf("You must specify the account name using --triton-account")
+	}
+	if d.PrivateKey == "" {
+		return fmt.Errorf("You must specify the SSH key using --triton-key")
+	}
+
+	if d.CloudApiURL != "coal" && strings.IndexAny(d.CloudApiURL, ":./") == -1 {
+		// Shortend format for the cloudapi name, e.g. "us-east-3b"
+		d.CloudApiURL = fmt.Sprintf("https://%s.%s", d.CloudApiURL, TritonDefaultCloudapiDomain)
+	}
+	return nil
+}
+
+// Start a host
+func (d *Driver) Start() error {
+	return fmt.Errorf("Start is not available for the triton driver")
+}
+
+// Stop a host gracefully
+func (d *Driver) Stop() error {
+	return fmt.Errorf("Stop is not available for the triton driver")
+}
+
+/* --------------------------------------------------------- */
+/* End of driver.Driver                                      */
+/* --------------------------------------------------------- */
+
+// GetHttpsURL returns a https url to the docker endpoint
+// e.g. https://1.2.3.4:2376
+func (d *Driver) GetHttpsURL() string {
+	return strings.Replace(d.DockerApiURL, "tcp://", "https://", 1)
+}
+
+/*
+ * Download the certificate authority file from the sdc-docker server.
+ */
+func (d *Driver) DownloadCa() error {
+	dockerHttpsUrl := d.GetHttpsURL()
+	caUrl := fmt.Sprintf("%s/ca.pem", dockerHttpsUrl)
+	log.Debugf("Downloading ca.pem file from %s", caUrl)
+
+	var resp *http.Response
+	var err error
+
+	if d.CloudApiURL == "coal" {
+		// Have to use an insecure https request for coal.
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
I'm thinking this could be necessary for other (non-coal) custom deployments of SDC. IMO this should be extracted to a flag (`--triton-tls-skip-verify` or something), which could perhaps default to `true` for the `coal` region.